### ISSUE TITLE
S905X QCA9377 WIFI Not Working

### ISSUE THREAD

**Issue by DiogoSilva48** [ophub/amlogic-s9xxx-armbian#98]
Hi, I have a S905X box with a QCA9377 Wireless Chip, the WIFI is not working on this box with this Armbian build, but it does work on the [LibreElec Test Build](https://test.libreelec.tv/LibreELEC-AMLGX.arm-11.0-nightly-20220131-71f8aac-box.img.gz), I know this box is some kind of rebranding but I don't know which, I have pictures of the board if you need. The QCA9377 should use ath10k driver and I have some error on dmesg (picture attached).
![image](https://user-images.githubusercontent.com/38466084/152037079-724c21d7-a9ad-4c5d-9a2c-39707312ee08.png)

**Comment by DiogoSilva48** [comment]
I think I tried an older version by mistake, I tried again with the 5.10 kernel and I have a different error, firmware, but I think I can fix it, I remember having this on other armbian builds a few years ago. Going to try and report.
![image](https://user-images.githubusercontent.com/38466084/152049783-fea7d516-84d9-4e50-9f2b-ac698908e00f.png)

**Comment by DiogoSilva48** [comment]
Don't know if I should close this or not, because I fixed it but I don't know if you want to include the firmware by default or not. Also, sorry for the spam.

**Comment by DiogoSilva48** [comment]
I still have to wait 5 minutes for "a start job is running for the raise network", before I can use Armbian or ssh into it, I tried a fix I found online, didn't work, any idea ? Thanks.

**Comment by ophub** [comment]
thank you for your sharing. When it is convenient, you can test the long-term updated kernels of 5.4/5.10/5.15.

If common, it can be added to the default firmware.

**Comment by DiogoSilva48** [comment]
Hi, thanks for answering. I found a much easier way to fix wifi. First, you do the usual "sudo apt update" and "sudo apt upgrade" and then "sudo armbian-config" and navigate to the option software and there's an option that says "install full firmware". After that, installs the required firmware to make wifi work, so might not be necessary to include by default : ). I'm going to try to switch kernels as well to see and update this.

**Comment by ophub** [comment]
OK, thanks for testing and sharing. Which `.dtb` and `u-boot` file did your box use? Does it support `writing to emmc`? The default values are recorded in `/etc/ophub-release`, if you customize them, update them.

You are the first friend who reported using this box, so you can find an online `product introduction link` and post it at your convenience. I add to the box list. thanks.

**Comment by DiogoSilva48** [comment]
I used the default .dtb and u-boot that and just worked fine. I can't use writting to emmc as of now because I think I broke my emmc, I bricked it many times before but I was always was able to fix it even after months, but this time I just can't get it to flash anything, I get errors flashing the u-boot I think and I don't have the stock firmware. Also, armbian even detect the emmc.  The box is [this one](https://www.tbee.com/en/tbee-box/), and this is what the board looks like.
![image](https://user-images.githubusercontent.com/38466084/152151296-648de7d9-122d-420f-8900-3968d5c9a2b3.png)
